.github/workflows/tests.yml

```diff
@@ -305,11 +305,29 @@ jobs:
         with:
           path: synapse
 
-      - name: Run actions/checkout@v2 for complement
-        uses: actions/checkout@v2
-        with:
-          repository: "matrix-org/complement"
-          path: complement
+      # Attempt to check out the same branch of Complement as the PR. If it
+      # doesn't exist, fallback to master.
+      - name: Checkout complement
+        shell: bash
+        run: |
+          mkdir -p complement
+          # Attempt to use the version of complement which best matches the current
+          # build. Depending on whether this is a PR or release, etc. we need to
+          # use different fallbacks.
+          #
+          # 1. First check if there's a similarly named branch (GITHUB_HEAD_REF
+          #    for pull requests, otherwise GITHUB_REF).
+          # 2. Attempt to use the base branch, e.g. when merging into release-vX.Y
+          #    (GITHUB_BASE_REF for pull requests).
+          # 3. Use the default complement branch ("master").
+          for BRANCH_NAME in "$GITHUB_HEAD_REF" "$GITHUB_BASE_REF" "${GITHUB_REF#refs/heads/}" "master"; do
+            # Skip empty branch names and merge commits.
+            if [[ -z "$BRANCH_NAME" || $BRANCH_NAME =~ ^refs/pull/.* ]]; then
+              continue
+            fi
+
+            (wget -O - https://github.com/matrix-org/complement/archive/$BRANCH_NAME.tar.gz | tar -xz --strip-components=1 -C complement) && break
```

Comment:
The URL needs to be quoted per shellcheck to avoid errors if the branch name somehow ends up containing a space

Reply:
Is there a reason to prefer wget over a shallow clone of the repo?

Reply:
I lost that somehow when refactoring... 😢

Reply:
I also don't know what you mean by "per spellcheck".

Reply:
Oh, sorry 🙂 Shellcheck is a really, really excellent static analysis tool for shell scripts: http://shellcheck.net/